src/Files.App/Views/BaseShellPage.cs

```diff
@@ -0,0 +1,739 @@
+			{
+				AssociatedTabInstance = this,
+				IsSearchResultPage = true,
+				SearchPathParam = FilesystemViewModel.WorkingDirectory,
+				SearchQuery = query,
+				SearchUnindexedItems = searchUnindexedItems,
+			});
+		}
+
+		public void NavigateWithArguments(Type sourcePageType, NavigationArguments navArgs)
+		{
+			NavigateToPath(navArgs.NavPathParam, sourcePageType, navArgs);
+		}
+
+		public void NavigateToPath(string navigationPath, NavigationArguments? navArgs = null)
+		{
+			NavigateToPath(navigationPath, FolderSettings.GetLayoutType(navigationPath), navArgs);
+		}
+
+		public Task TabItemDragOver(object sender, DragEventArgs e)
+		{
+			return SlimContentPage?.CommandsViewModel.CommandsModel.DragOver(e);
+		}
+
+		public Task TabItemDrop(object sender, DragEventArgs e)
+		{
+			return SlimContentPage?.CommandsViewModel.CommandsModel.Drop(e);
+		}
+
+		public async void Refresh_Click()
+		{
+			if (InstanceViewModel.IsPageTypeSearchResults)
+			{
+				ToolbarViewModel.CanRefresh = false;
+				var searchInstance = new FolderSearch
+				{
+					Query = InstanceViewModel.CurrentSearchQuery ?? (string)TabItemArguments.NavigationArg,
+					Folder = FilesystemViewModel.WorkingDirectory,
+					ThumbnailSize = InstanceViewModel.FolderSettings.GetIconSize(),
+					SearchUnindexedItems = InstanceViewModel.SearchedUnindexedItems
+				};
+				await FilesystemViewModel.SearchAsync(searchInstance);
+			}
+			else if (CurrentPageType != typeof(WidgetsPage))
+			{
+				ToolbarViewModel.CanRefresh = false;
+				FilesystemViewModel?.RefreshItems(null);
+			}
+		}
+
+		public virtual void Back_Click()
+		{
+			var previousPageContent = ItemDisplay.BackStack[ItemDisplay.BackStack.Count - 1];
+			var previousPageNavPath = previousPageContent.Parameter as NavigationArguments;
+			previousPageNavPath.IsLayoutSwitch = false;
+			if (previousPageContent.SourcePageType != typeof(WidgetsPage))
+				InstanceViewModel.FolderSettings.GetLayoutType(previousPageNavPath.IsSearchResultPage ? previousPageNavPath.SearchPathParam : previousPageNavPath.NavPathParam); // Update layout type
+			SelectSidebarItemFromPath(previousPageContent.SourcePageType);
+
+			if (previousPageContent.SourcePageType == typeof(WidgetsPage))
+				ItemDisplay.GoBack(new EntranceNavigationTransitionInfo());
+			else
+				ItemDisplay.GoBack();
+		}
+
+		public virtual void Forward_Click()
+		{
+			var incomingPageContent = ItemDisplay.ForwardStack[ItemDisplay.ForwardStack.Count - 1];
+			var incomingPageNavPath = incomingPageContent.Parameter as NavigationArguments;
+			incomingPageNavPath.IsLayoutSwitch = false;
+			if (incomingPageContent.SourcePageType != typeof(WidgetsPage)) // Update layout type
+				InstanceViewModel.FolderSettings.GetLayoutType(incomingPageNavPath.IsSearchResultPage ? incomingPageNavPath.SearchPathParam : incomingPageNavPath.NavPathParam);
+			SelectSidebarItemFromPath(incomingPageContent.SourcePageType);
+			ItemDisplay.GoForward();
+		}
```

Comment:
Those two methods could be refined to use the same logic to fetch the layout type. Only the first line and the last few lines would be different in the Forward and Backward.